collector/textfile.go

```diff
@@ -52,31 +65,165 @@ func NewTextFileCollector() (Collector, error) {
 	c := &textFileCollector{
 		path: *textFileDirectory,
 	}
+	return c, nil
+}
 
-	if c.path == "" {
-		// This collector is enabled by default, so do not fail if
-		// the flag is not passed.
-		log.Infof("No directory specified, see --collector.textfile.directory")
-	} else {
-		textFileAddOnce.Do(func() {
-			prometheus.DefaultGatherer = prometheus.Gatherers{
-				prometheus.DefaultGatherer,
-				prometheus.GathererFunc(func() ([]*dto.MetricFamily, error) { return c.parseTextFiles(), nil }),
+func convertMetricFamilies(metricFamilies []*dto.MetricFamily, ch chan<- prometheus.Metric) {
+	var valType prometheus.ValueType
+	var val float64
+
+	for _, mf := range metricFamilies {
+		labelNames := map[string]struct{}{}
+		for _, metric := range mf.Metric {
+			labelPairs := metric.GetLabel()
+			for _, label := range labelPairs {
+				if _, ok := labelNames[label.GetName()]; !ok {
+					labelNames[label.GetName()] = struct{}{}
+				}
 			}
-		})
+		}
+
+		for _, metric := range mf.Metric {
+			labelPairs := metric.GetLabel()
+			var labels []string
+			var labelVals []string
```

Comment:
Here I would suggest `names` and `values`, or if you want to be verbose, `labelNames` / `labelValues` (then you'll want to change the existing `labelNames` in the outer loop into `allLabelNames` or `mfLabelNames`.